contract/models/contract_recurrency_mixin.py

```diff
@@ -93,9 +93,16 @@ class ContractRecurrencyMixin(models.AbstractModel):
         string="Last Date Invoiced", readonly=True, copy=False
     )
 
-    @api.depends("next_period_date_start")
+    @api.depends(
+        "next_period_date_start",
+        "recurring_invoicing_type",
+        "recurring_rule_type",
+        "recurring_interval",
+    )
```

Comment:
I'm wondering why all params send to the `get_next_invoice_date` are not present in this list?

should we add those ones, but I miss a bit of context here maybe we don't expect those fields update  while this method is to be called:
```suggestion
        "next_period_date_start",
        "recurring_invoicing_type", 
        "recurring_invoicing_offset",
        "recurring_rule_type",
        "recurring_interval",
       "date_end",
    )
```